### PR TITLE
`ipAsset` integration tests - edge and complex cases

### PR DESCRIPTION
**Pull request by bpolania**
## Description

This PR adds edge cases and more complex tests to the `ipAsset` module integration tests.

### CODE REVIEW

File: packages/core-sdk/test/integration/ipAsset.test.ts [hunk 1]
```diff
@@ -507,6 +608,300 @@ describe("IP Asset Functions", () => {
...
+            },
+          ],
+          ipMetadata: {
+            ipMetadataURI: "test-uri",
+            ipMetadataHash: toHex("test-metadata-hash", { size: 32 }),
+            nftMetadataHash: toHex("test-nft-metadata-hash", { size: 32 }),
+          },
+          royaltyShares: [
+            {
+              recipient: process.env.TEST_WALLET_ADDRESS! as Address,
```
[the remark this answers is not in the document]
Comment (bpolania) [reply packages/core-sdk/test/integration/ipAsset.test.ts:657]: Fixed

File: packages/core-sdk/test/integration/ipAsset.test.ts [hunk 2]
```diff
@@ -1217,4 +1612,191 @@ describe("IP Asset Functions", () => {
...
     });
   });
+
+  describe("Other Edge Cases", () => {
+    let client: StoryClient;
+    let noCommercialLicenseTermsId: bigint;
+    let parentIpId: Hex;
+
+    before(async () => {
+      client = getStoryClient();
```
[the remark this answers is not in the document]
Comment (bpolania) [reply packages/core-sdk/test/integration/ipAsset.test.ts:1622]: No. I needed to reset the scope to force some errors.

File: packages/core-sdk/test/integration/ipAsset.test.ts [hunk 3]
```diff
@@ -1217,4 +1612,191 @@ describe("IP Asset Functions", () => {
...
+      const res = await client.license.registerNonComSocialRemixingPIL({
+        txOptions: { waitForTransaction: true },
+      });
+      noCommercialLicenseTermsId = res.licenseTermsId!;
```
[the remark this answers is not in the document]
Comment (bpolania) [reply packages/core-sdk/test/integration/ipAsset.test.ts:1626]: Ideally, we should try, but in some cases, since we are trying multiple configurations, they need to be reset anyway. In other cases, they are out of scope.

File: packages/core-sdk/test/integration/ipAsset.test.ts [hunk 4]
```diff
@@ -1217,4 +1612,191 @@ describe("IP Asset Functions", () => {
...
+
+      const tokenId = await getTokenId();
+      const parentIpResponse = await client.ipAsset.register({
+        nftContract: mockERC721,
+        tokenId: tokenId!,
+        txOptions: { waitForTransaction: true },
+      });
+      parentIpId = parentIpResponse.ipId!;
```
[the remark this answers is not in the document]
Comment (bpolania) [reply packages/core-sdk/test/integration/ipAsset.test.ts:1634]: The previous `parentIpId` is out of scope, i.e., not in the same `describe`

File: packages/core-sdk/test/integration/ipAsset.test.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (DracoLi) [reply packages/core-sdk/test/integration/ipAsset.test.ts:1626]: in this case, shouldn't `noCommercialLicenseTermsId` be the same ID as the one you have in the outer scope? Since the terms are the same, the POC should return the same license id i believe. So we can avoid doing multical rpc calls.

File: packages/core-sdk/test/integration/ipAsset.test.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (DracoLi) [reply packages/core-sdk/test/integration/ipAsset.test.ts:1634]: we do have a `parentIpId` in the most outer scope `IP Asset Functions`. I understand if we need a brand new IP with a clean state. But if we don't or can avoid it, then lets try to reuse.

**Review by DracoLi — APPROVED** [review]
Tests looks great. my comments on reducing RPC usage if possible is not blocking but a general item we should be considering whenever we add more tests.
